src/chocolatey.resources/helpers/ChocolateyTabExpansion.ps1

```diff
@@ -35,56 +35,94 @@ function script:chocoCmdOperations($commands, $command, $filter, $currentArgumen
 
 try {
-    # if license exists
-    # add in pro/biz switches
+    $licenseFile = Get-Item -Path "$env:ChocolateyInstall\license\chocolatey.license.xml" -ErrorAction Ignore
+
+    if ($licenseFile) {
+        # Add pro-only commands
+        $script:chocoCommands = @(
+            $script:chocoCommands
+            'download'
+            'optimize'
+        )
+
+        $commandOptions.download = "--internalize --internalize-all-urls --ignore-dependencies --installed-packages --ignore-unfound-packages --resources-location='' --download-location='' --outputdirectory='' --source='' --version='' --prerelease --user='' --password='' --cert='' --certpassword='' --append-use-original-location --recompile --disable-package-repository-optimizations"
+        $commandOptions.sync = "--output-directory='' --id='' --package-id=''"
+        $commandOptions.optimize = "--deflate-nupkg-only --id=''"
+
+        # Add pro switches to commands that have additional switches on Pro
+        $proInstallUpgradeOptions = " --install-directory='' --package-parameters-sensitive='' --max-download-rate='' --install-arguments-sensitive='' --skip-download-cache --use-download-cache --skip-virus-check --virus-check --virus-positives-minimum='' --deflate-package-size --no-deflate-package-size --deflate-nupkg-only"
+
+        $commandOptions.install += $proInstallUpgradeOptions
+        $commandOptions.upgrade += $proInstallUpgradeOptions + " --exclude-chocolatey-packages-during-upgrade-all --include-chocolatey-packages-during-upgrade-all"
+        $commandOptions.new += " --build-package --use-original-location --keep-remote --url='' --url64='' --checksum='' --checksum64='' --checksumtype='' --pause-on-error"
+        $commandOptions.pin += " --note=''"
+
+        # Add Business-only commands and options if the license is a Business or Trial license
+        [xml]$xml = Get-Content -Path $licenseFile.FullName
+        $licenseType = $xml.license.type
+
+        if ('Business', 'BusinessTrial' -contains $licenseType) {
+
+            # Add business-only commands
+            $script:chocoCommands = @(
+                $script:chocoCommands
+                'sync'
+            )
+
+            $commandOptions.list += " --audit"
+            $commandOptions.uninstall += " --from-programs-and-features"
+            $commandOptions.new += " --file='' --file64='' --from-programs-and-features --remove-architecture-from-name --include-architecture-in-name"
+
+            # Add --use-self-service to commands that support it
+            $selfServiceCommands = 'list', 'find', 'search', 'info', 'install', 'upgrade', 'uninstall', 'pin', 'outdated', 'push', 'download', 'sync', 'optimize'
+            foreach ($command in $selfServiceCommands) {
+                $commandOptions.$command += ' --use-self-service'
+            }
+        }
+    }
 }
 catch {
 }
 
+foreach ($key in $commandOptions.Keys) {
+    $commandOptions.$key += $allcommands
+}

```

Comment:
This doesn't seem to work. And I haven't been able to find a way that does work... 
![image](https://user-images.githubusercontent.com/30301021/230737888-58c5fa95-a037-4e26-b01a-e32b19586a70.png)

Reply:
Ah yep, classic footgun there, love those. I've amended the commit to collect the keys in a separate array before going and changing the hashtable.